### PR TITLE
feat: add pull requests endpoint under v1-alpha

### PR DESCRIPTION
**Pull request by epipav**
## What

Adds `GET /v1-alpha/projects/{slug}/development/pull-requests`, the port of the Nuxt pull-requests widget handler (`frontend/server/api/widget/development/pull-requests.get.ts`) to the api package.

Query: `repos[]`, `startDate`, `endDate` (UTC calendar days, `YYYY-MM-DD`) and `granularity` (`daily | weekly | monthly | quarterly | yearly`, required).

Response:

- `openedSummary`, `mergedSummary`, `closedSummary`: `PeriodSummary` blocks (current and previous counts, signed `percentageChange`, `changeValue`, ISO `periodFrom`/`periodTo`)
- `avgResolveTimeSeconds`: average time to resolve a pull request in the period, in seconds; `null` when nothing was resolved
- `data[]`: one `{ startDate, endDate, open, merged, closed }` per granularity bucket, dates as ISO-8601 UTC

## Changes against the Nuxt handler

- The top-level `summary` is dropped; it duplicated `openedSummary`.
- `avgVelocityInDays` is renamed `avgResolveTimeSeconds`; the pipe value was already in seconds.
- `percentageChange` is signed (Nuxt used `Math.abs`).
- Summary and bucket dates are ISO-8601 UTC; Nuxt returned `''` and bare `YYYY-MM-DD`.
- An unknown slug returns 200 with zero summaries and an empty series, per the epic's definition of done; Nuxt returned 404.
- Every Tinybird failure, the bucket lookup included, returns 503 `upstream_unavailable`; Tinybird's own status never reaches the caller.
- An empty `repos=` is dropped instead of being forwarded as a filter that matches nothing (Ajv coerces it to `['']`); Nuxt dropped it too.

## Decisions

- **Bucket resolved once per request.** The Tinybird client resolves a `bucketId` before every query that carries `project`, so a naive port would make 20 HTTP calls. The route calls `getBucketIdForProject` once and passes the id to the ten pipe queries, which run concurrently with `Promise.all` as in Nuxt. A `null` bucket is the unknown-slug case. No response caching in this PR (T-090 load test tracks this endpoint); the route description states the 11-call cost.
- **`granularity` is required.** Without one the `activities_count` pipe returns a single summary row instead of a series, so a missing value is a 400. This matches IN-1331, IN-1332 and IN-1334.
- **The resolved current range is always sent.** With dates omitted the route sends `2010-01-01` through today, so `periodFrom`/`periodTo` describe the range that was queried.
- **Bucket key stays `open`.** The ticket lists the renames explicitly; the field description states it is the count of pull requests opened in the bucket. Renaming it to `opened` is a candidate for the `/v1` graduation review.
- **Series merged by bucket start over their union.** Whether the pipe emits a row for an empty bucket is undocumented, so `data` covers every bucket start any of the three series reports, matched by the normalised start instant and ordered by it; a count missing from a series is 0. Nuxt anchored on the opened series and joined by index, which could misplace or drop merged and closed activity.

Helpers (Tinybird date formatting, ISO conversion, granularity query schema, 503 wrapper, bucket-once fan-out) live in the route file per the parallel-PR rules; the consolidation chore after the five Development PRs merge moves them to a shared module.

## Tests

`api/tests/development-pull-requests.test.ts`: 33 cases, all green. Stubs global `fetch` at the HTTP boundary and routes by pipe, so the real Tinybird client builds the URLs, resolves the bucket and classifies the errors. Covers the exact 11 calls and their params, concurrency, the repos filter, defaults, unknown slug, empty data, validation (400), every Tinybird failure class (503), Cache-Control, the OpenAPI entry (tag, descriptions, nullable, required granularity) and field stripping.

`pnpm test`, `pnpm tsc-check`, `pnpm lint` and `pnpm format:check` pass in `api/`.

Includes the routing-test parser fix from the base branch (62b136b9).

Based on main, which includes #2241 (IN-1348, directory-based route registration) as dd37a7a9. Jira: IN-1333

- `main` 
  - \#2241
    - **feat: add pull requests endpoint under v1-alpha** :point\_left:

### CODE REVIEW

File: api/src/versions/v1-alpha/development/pull-requests.ts [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+    // The three series share their bucket bounds, so they join by index
+    // (see frontend/server/data/tinybird/development/pull-requests.ts).
+    data: rows.openedSeries.map((row, index) => ({
+      startDate: toIsoUtc(row.startDate),
+      endDate: toIsoUtc(row.endDate),
+      open: row.activityCount ?? 0,
+      merged: rows.mergedSeries[index]?.activityCount ?? 0,
+      closed: rows.closedSeries[index]?.activityCount ?? 0,
+    })),
+  };
```
[the remark this answers is not in the document]
Comment (gaspergrom) [reply api/src/versions/v1-alpha/development/pull-requests.ts:271]: This got fixed in a later commit that keys the merge by startDate instead of array index.

File: api/src/versions/v1-alpha/development/pull-requests.ts [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+    closedSummary: summarize(rows.closedCurrent, rows.closedPrevious),
+    avgResolveTimeSeconds: rows.velocity[0]?.averagePullRequestResolveVelocitySeconds ?? null,
+    data: rows.openedSeries.map((row) => {
+      const startDate = toIsoUtc(row.startDate);
+      return {
+        startDate,
+        endDate: toIsoUtc(row.endDate),
+        open: row.activityCount ?? 0,
+        merged: mergedByBucket.get(startDate) ?? 0,
+        closed: closedByBucket.get(startDate) ?? 0,
```
[the remark this answers is not in the document]
Comment (gaspergrom) [reply api/src/versions/v1-alpha/development/pull-requests.ts]: Also fixed in a later commit. Each series merges independently into the bucket map now, so the output covers the union of all three.

**Review by gaspergrom — APPROVED** [review]
Traced the bucket-merge logic and the Tinybird failure mapping against the tests, both hold up. Good coverage on the union-merge edge cases and the concurrency check.

**Review by themarolt — APPROVED** [review]
lgtm - the bucket-once fan-out and the tinybird date formatting line up with the nuxt handler the way the description says, nothing from me
